api/api/serializers/media_serializers.py

```diff
@@ -334,6 +295,118 @@ def needs_db(self) -> bool:
         return False
 
 
+class MediaSearchRequestSerializer(MediaListRequestSerializer):
+    """This serializer parses and validates search query string parameters."""
+
+    fields_names = [
+        *MediaListRequestSerializer.fields_names,
+        "q",
+        "creator",
+        "tags",
+        "title",
+    ]
+    """
+    Keep the fields names in sync with the actual fields below as this list is
+    used to generate Swagger documentation.
+    """
+
+    q = serializers.CharField(
+        label="query",
+        help_text="A query string that should not exceed 200 characters in length",
+        required=False,
+    )
+    creator = serializers.CharField(
+        label="creator",
+        help_text="Search by creator only. Cannot be used with `q`.",
+        required=False,
+        max_length=200,
+    )
+    tags = serializers.CharField(
+        label="tags",
+        help_text="Search by tag only. Cannot be used with `q`.",
+        required=False,
+        max_length=200,
+    )
+    title = serializers.CharField(
+        label="title",
+        help_text="Search by title only. Cannot be used with `q`.",
+        required=False,
+        max_length=200,
+    )
+
+    def validate_q(self, value):
+        return self._truncate(value)
+
+    def validate_creator(self, value):
+        return self._truncate(value)
+
+    def validate_tags(self, value):
+        return self._truncate(value)
+
+    def validate_title(self, value):
+        return self._truncate(value)
+
+
+@extend_schema_serializer(
+    exclude_fields=[
+        "unstable__sort_by",
+        "unstable__sort_dir",
+        "unstable__authority",
+        "unstable__authority_boost",
+        "unstable__include_sensitive_results",
+        "internal__index",
+        # These are path parameters, not query parameters.
+        "source",
```

Comment:
What is the effect of this? I can still use these as query parameters. E.g.: `http://localhost:50280/v1/audio/source/jamendo/?tag=peaceful` works and filters as expected.

Reply:
This part specifically only removes them from the documentation. The fields need to not exist on the serializer for them to not appear in the serializer data. I've shared a proposal in another comment for a way to untangle this a bit and prevent this confusing cross-pollination of parameters. It prevents us from needing to modify the documentation as well.